templates/types/streaming/nextjs/app/components/ui/chat/index.ts

```diff
@@ -28,6 +29,23 @@ export type CsvData = {
   csvFiles: CsvFile[];
 };
 
+export type TextEmbedding = {
+  text: string;
+  embedding: number[];
+};
+
+export type PdfFile = {
```

Comment:
how about having `File` and having an `ContentFile` that extends `File` with `embeddings`?